libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -294,9 +296,12 @@ namespace eosio { namespace chain {
          void set_greylist_limit( uint32_t limit );
          uint32_t get_greylist_limit()const;
 
-         void add_to_ram_correction( account_name account, uint64_t ram_bytes );
+         void add_to_ram_correction( account_name account, uint64_t ram_bytes, uint32_t action_id, const char* event_id );
```

Comment:
Since ram correction are generating a different OP (`RAM_CORRECTION_OP`) than normal ram operation (`RAM_OP`), most of the fields would not be used here. Hence why I decided to not use a `ram_trace` for this one.